### ISSUE TITLE
Better management of coverage layers

### ISSUE THREAD

**Issue by zacharlie** [kartoza/geodata-mart#93]
[Currently](https://github.com/kartoza/geodata-mart/commit/42ed5b9754daac830f3bf5ddf03ee244fc5c0c5c), a project coverage is defined by a simple MultiPolygon field provided by GeoDjango. A "Project Coverage File" (which is limited to a *single* file per project, .i.e. no shp/dbf) model provides a way to upload a layer, including an "on save" hook/ signal that will attempt to convert the file (if it is an OGR data source with Multipolygon type) into a valid coverage for the project.

The GeoDjango "LayerMapping" is not a suitable approach for gathering "just" the geometry and applying it to an existing record.

No user message is generated on error (simply logs to console), and only the first feature is used.

Basically this requires the input to be a valid, single feature multipolygon geojson file, and there is no validation on this during upload.

A more robust mechanism, such as unzipping a folder and processing it, or supporting multiple geometry types, would be needed in the future.
